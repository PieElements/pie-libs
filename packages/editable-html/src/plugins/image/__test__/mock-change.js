import { Data } from 'slate';
import { stub } from 'sinon';

export default function MockChange() {

  this.setNodeByKey = stub().returns(this);

  this.removeNodeByKey = stub().returns(this);
  this.insertInline = stub().returns(this);
}

export function MockDocument() {

  this.getChild = stub().returns({ data: Data.create({}) });

  this.getDescendant = stub();
}