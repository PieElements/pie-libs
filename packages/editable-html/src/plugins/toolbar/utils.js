export const findSingleNode = (value) => {

  if (!value || !value.isCollapsed || !value.startKey) {
    return;
  }

  const inline = value.document.getClosestInline(value.startKey);

  if (inline) {
    return inline;
  }

  const block = value.document.getClosestBlock(value.startKey);

  if (block) {
    return block;
  }

}