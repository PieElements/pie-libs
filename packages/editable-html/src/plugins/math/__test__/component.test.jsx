import React from 'react';
import renderer from 'react-test-renderer';
import { classObject, mockMathInput } from '../../../__test__/utils';
import debug from 'debug';
const log = debug('editable-html:test:math');

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

beforeAll(() => {
  configure({ adapter: new Adapter() });
});

mockMathInput();

describe('component', () => {

  let mod, MathComponent, node, editor, classes, mathWrapper;

  mathWrapper = {
    change: jest.fn()
  }

  beforeEach(() => {

    try {
      mod = require('../component.jsx');
    } catch (e) {
      log(e);
    }

    MathComponent = mod.MathComponent;
    classes = classObject('root', 'selected');
    editor = {
      value: {
        isFocused: true
      },
      change: jest.fn()
    }

    node = {
      data: {
        get: jest.fn(() => 'latex')
      }
    }

  });

  test('snapshot', () => {
    const tree = renderer.create(<MathComponent
      classes={classes}
      editor={editor}
      node={node}
    />);

    expect(tree).toMatchSnapshot();
  });

  xdescribe('componentDidUpdate', () => {

    //TODO: Testing components w/ refs is a pain in the ....
    //Isolate the use of refs to only the wrapper arount the MathInput
    //So that tests can be written.
  });
});