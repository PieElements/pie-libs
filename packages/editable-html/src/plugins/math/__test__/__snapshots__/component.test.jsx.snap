// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component snapshot 1`] = `
<div
  className="root"
>
  <div
    data-mock-component="true"
  >
    EditableMathInput
  </div>
</div>
`;
